core/src/main/java/org/elasticsearch/index/query/PrefixQueryBuilder.java

```diff
@@ -71,4 +160,61 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String getName() {
         return NAME;
     }
+
+    @Override
+    protected Query doToQuery(QueryParseContext parseContext) throws IOException {
+        MultiTermQuery.RewriteMethod method = QueryParsers.parseRewriteMethod(rewrite, null);
+
+        Query query = null;
+        MappedFieldType fieldType = parseContext.fieldMapper(fieldName);
+        if (fieldType != null) {
+            query = fieldType.prefixQuery(value, method, parseContext);
+        }
+        if (query == null) {
+            PrefixQuery prefixQuery = new PrefixQuery(new Term(fieldName, BytesRefs.toBytesRef(value)));
+            if (method != null) {
+                prefixQuery.setRewriteMethod(method);
+            }
+            query = prefixQuery;
+        }
+        
+        return query;
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        if (Strings.isEmpty(this.fieldName)) {
+            validationException = addValidationError("field name cannot be null or empty.", validationException);
+        }
+        if (this.value == null) {
+            validationException = addValidationError("query text cannot be null", validationException);
+        }
+        return validationException;
+    }
+
+    @Override
+    protected PrefixQueryBuilder doReadFrom(StreamInput in) throws IOException {
+        PrefixQueryBuilder prefixQueryBuilder = new PrefixQueryBuilder(in.readString(), in.readGenericValue());
+        prefixQueryBuilder.rewrite = in.readOptionalString();
+        return prefixQueryBuilder;
+    }
+
+    @Override
+    protected void doWriteTo(StreamOutput out) throws IOException {
+        out.writeString(fieldName);
+        out.writeGenericValue(value);
+        out.writeOptionalString(fieldName);
+    }
+
+    @Override
+    protected final int doHashCode() {
+        return Objects.hash(getClass(), fieldName, value);
```

Comment:
Not sure exactly at what time I wrote this, but maybe that could explain how I missed this. Thanks!